packages/@lwc/engine-core/src/patches/detect-synthetic-cross-root-aria.ts

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import {
+    assign,
+    globalThis,
+    isTrue,
+    KEY__NATIVE_GET_ELEMENT_BY_ID,
+    KEY__NATIVE_QUERY_SELECTOR_ALL,
+    isNull,
+    isUndefined,
+    getOwnPropertyDescriptor,
+    defineProperty,
+    ID_REFERENCING_ATTRIBUTES_SET,
+    isString,
+    isFunction,
+    StringSplit,
+    ArrayFilter,
+    hasOwnProperty,
+    KEY__SHADOW_TOKEN,
+} from '@lwc/shared';
+import { onReportingEnabled, report, ReportId } from '../framework/reporting';
+import { getAssociatedVMIfPresent, VM } from '../framework/vm';
+import { logWarn } from '../shared/logger';
+
+//
+// The goal of this code is to detect invalid cross-root ARIA references in synthetic shadow DOM.
+// These invalid references should be fixed before the offending components can be migrated to native shadow DOM.
+// When invalid usage is detected, we warn in dev mode and call the reporting API if enabled.
+// See: https://lwc.dev/guide/accessibility#link-ids-and-aria-attributes-from-different-templates
+//
+
+// Use the unpatched native getElementById/querySelectorAll rather than the synthetic one
+const getElementById = globalThis[KEY__NATIVE_GET_ELEMENT_BY_ID] as typeof document.getElementById;
+
+const querySelectorAll = globalThis[
+    KEY__NATIVE_QUERY_SELECTOR_ALL
+] as typeof document.querySelectorAll;
```

Comment:
Unfortunately we need to get the native `getElementById` and `querySelectorAll`. These are overridden by synthetic shadow, so the only way to pass them from synthetic shadow to engine-dom is to expose them somewhere else.

My assumption is that Locker will block access to these.